barman/server.py

```diff
@@ -797,10 +831,109 @@ def check_postgres(self, check_strategy):
+             connection.
+
+        :param CheckStrategy check_strategy: The strategy for the management
+            of the result of this check
+        :param dict[str, None|str] remote_status: Remote status information used
+            by this check
+        :param str|None suffix: A suffix to be appended to the check name
+        """
+        # If we have wal-specific conninfo then we must use those to get
+        # the remote status information for the check
+        if self.config.wal_streaming_conninfo is not None:
+            streaming_conninfo, conninfo = self.config.get_wal_conninfo()
+            with closing(StreamingConnection(streaming_conninfo)) as streaming, closing(
+                PostgreSQLConnection(conninfo, slot_name=self.config.slot_name)
+            ) as postgres:
+                remote_status = postgres.get_remote_status()
+                remote_status.update(streaming.get_remote_status())
+
+                self._check_has_monitoring_privileges(
+                    check_strategy, remote_status, "WAL streaming"
+                )
+                self._check_streaming_supported(
+                    check_strategy, remote_status, "WAL streaming"
+                )
+                self._check_wal_level(check_strategy, remote_status, "WAL streaming")
+                self.check_identity(check_strategy, remote_status, "WAL streaming")
+                self._check_replication_slot(
+                    check_strategy, remote_status, "WAL streaming"
+                )
+        else:
+            # Use the status for the existing postgres connections
+            remote_status = self.get_remote_status()
+            self._check_replication_slot(check_strategy, remote_status)
+
+    def _check_replication_slot(self, check_strategy, remote_status=None, suffix=None):
+        """
+        Check the replication slot used for WAL streaming.
+
+        If ``streaming_archiver`` is enabled, checks that the replication slot specified
+        in the configuration exists, is initialised and is active.
+
+        If ``streaming_archiver`` is disabled, checks that the replication slot does not
+        exist.
+
+        :param CheckStrategy check_strategy: The strategy for the management
+            of the result of this check
+        :param dict[str, None|str] remote_status: Remote status information used
+            by this check
+        :param str|None suffix: A suffix to be appended to the check name
+        """
         # Check the presence and the status of the configured replication slot
         # This check will be skipped if `slot_name` is undefined
         if self.config.slot_name:
-            check_strategy.init_check("replication slot")
+            check_name = "replication slot" + ("" if suffix is None else f" ({suffix})")
+            check_strategy.init_check(check_name)
+            if remote_status is None:
+                remote_status = self.get_remote_status()
```

Comment:
Ok I checked the history and this logic wasn't even present in the original so we can indeed remove this. I'll also remove the `None` default since omitting `remote_status` is not a valid way of calling this function.